src/GitHubApps/src/DarcBot/AzureFunction.cs

```diff
@@ -0,0 +1,212 @@
+        private static Regex _darcBotPropertyRegex = new Regex(@"\[(?<key>.+)=(?<value>[^\]]+)\]", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
+        private static readonly string docLink = "[DarcBot documentation](https://github.com/dotnet/arcade-services/tree/master/src/GitHubApps/src/DarcBot/Readme.md)";
+
+        [FunctionName("GitHubWebHook")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "POST", Route = null)]
+            HttpRequestMessage req, ILogger log)
+        {
+            _log = log;
+
+            dynamic data = await req.Content.ReadAsAsync<object>();
+
+            string userType = data?.comment?.user?.type;
+            if (userType == "Bot")
+            {
+                _log.LogInformation("Comment is from DarcBot, ignoring.");
+                return new OkObjectResult($"Ignoring DarcBot comment");
+            }
+            string gitHubAction = data?.action;
+
+            int installationId = data?.installation?.id;
+            long repositoryId = data?.repository?.id;
+            int prNumber = data?.issue?.number;
+
+            // Check for darcbot directed comments
+            if (gitHubAction == "created")
+            {
+                /* Nothing to do, this is the section where you would handle user comments if they contained explicit directives to darcbot.
+                   Note: be sure to ignore comments from issue?.user?.type == "Bot" so you don't get stuck in a payload delivery loop */
+            }
+
+            if (gitHubAction != "opened" &&
+                gitHubAction != "reopened" &&
+                gitHubAction != "closed")
+            {
+                _log.LogInformation($"Received github action '{gitHubAction}', nothing to do");
+                return new OkObjectResult($"DarcBot has nothing to do with github issue action '{gitHubAction}'");
+            }
+
+            // Determine identifiable information for triage item
+            string issueBody = data?.issue?.body;
+            TriageItem triageItem = GetTriageItemProperties(issueBody);
+
+            string issueUrl = data?.issue?.html_url;
+            triageItem.Url = issueUrl;
+
+            if (triageItem == null)
+            {
+                /* Item is not a triage item (does not contain identifiable information), do nothing */
+                _log.LogInformation($"{data?.issue?.url} is not a triage type issue.");
+                return new OkObjectResult("No identifiable information detected");
+            }
+
+            IGitHubClient gitHubClient = new DarcBotGitHubClient(installationId, _log);
+
+            if (gitHubAction == "opened" ||
+                gitHubAction == "reopened")
+            {
+                
+                // First, look for duplicate issues that are open
+                var openIssues = new RepositoryIssueRequest
```

Comment:
Could we apply some label to these, so we can only find thing with that particular label? Rather than searching all issues?

Reply:
it's just the open issues, but having a label for these seems like a reasonable suggestion.  I'll think about that one.  Thanks!